proposals/lip-0057.md

```diff
@@ -1910,6 +1916,46 @@ After the transactions in a block `b` are executed, the properties related to mi
 ```python
 def afterTransactionsExecute(b: Block) -> None:
     height = b.header.height
+    
+    updateProductivity(b.header)
+
+    # if not last block of round no further updates are needed.
+    if isEndOfRound(height) == False:
+        return
+
+    # Block b is an end-of-round block. Need to update snapshot and select validators for next round.
+    # This must be done after the properties related to missed blocks are updated.
+    
+    # No updates in validators and BFT weights are performed during bootstrap period.
+    if roundNumber(height+1) <= genesisDataStore.initRounds:
```

Comment:
As we are refactoring, can we also improve the function naming. `roundNumber` is not a good name and previously was both used as function name and for local variables. I would suggest `getRoundNumber` or `getRoundNumberFromHeight`.

Reply:
Further, I would suggest to once define a local variable
`currentRound = roundNumber(height)` instead of using several function calls here. Then, I would write `currentRound + 1` instead of `roundNumber(height+1)` (I assume this is the same as `isEndOfRound(height) == True`). I would find this more clear.

Reply:
Good point. I refactored and used also `nextRound = roundNumber(height+1)` which is also used in the implementation, for clarity. Have a look. Of course this can be skipped by replacing `nextRound` by `currentRound+1`.

Reply:
Looks good.